modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsa.ts

```diff
@@ -196,8 +268,102 @@ export class EcdsaUtils extends baseTSSUtils<KeyShare> {
       enterprise: enterprise,
     };
 
-    const keychains = await this.baseCoin.keychains().add(createBitGoMPCParams);
-    return keychains;
+    return await this.baseCoin.keychains().add(createBitGoMPCParams);
+  }
+
+  async getBackupEncryptedNShare(
+    backupShare: BackupKeyShare,
+    recipientIndex: number,
+    recipientGpgPublicArmor: string,
+    isThirdPartyBackup = false
+  ): Promise<EncryptedNShare> {
+    let backupToBitgoShare: EncryptedNShare;
+    if (isThirdPartyBackup && backupShare.bitGoHeldKeyShares?.keyShares) {
+      const backupToOtherApiShare = backupShare.bitGoHeldKeyShares.keyShares.find(
+        (keyShare) => keyShare.from === 'backup' && keyShare.to === getParticipantFromIndex(recipientIndex)
+      );
+      if (!backupToOtherApiShare) {
+        throw new Error('Missing Backup to Bitgo key share');
+      }
+      backupToBitgoShare = await encryptNShareFromAPIKeyShare(backupToOtherApiShare, recipientGpgPublicArmor);
+    } else {
+      assert(backupShare.userHeldKeyShare);
+      backupToBitgoShare = await encryptNShare(backupShare.userHeldKeyShare, recipientIndex, recipientGpgPublicArmor);
+    }
+    return backupToBitgoShare;
+  }
+
+  /** @inheritdoc */
+  async createUserKeychainFromThirdPartyBackup(
+    userGpgKey: openpgp.SerializedKeyPair<string>,
+    userKeyShare: KeyShare,
+    backupKeyShares: ApiKeyShare[],
+    bitgoKeychain: Keychain,
+    passphrase: string,
+    originalPasscodeEncryptionCode?: string
+  ): Promise<Keychain> {
+    const bitgoKeyShares = bitgoKeychain.keyShares;
+    if (!bitgoKeyShares) {
+      throw new Error('Missing BitGo key shares');
+    }
+    if (!bitgoKeychain.commonKeychain) {
+      throw new Error(`Missing common key chain: ${bitgoKeychain.commonKeychain}`);
+    }
+
+    const bitGoToUserShare = bitgoKeyShares.find((keyShare) => keyShare.from === 'bitgo' && keyShare.to === 'user');
+    if (!bitGoToUserShare) {
+      throw new Error('Missing BitGo to User key share');
+    }
+
+    const backupToUserShare = bitgoKeyShares.find((keyShare) => keyShare.from === 'backup' && keyShare.to === 'user');
+    if (!backupToUserShare) {
+      throw new Error('Missing BitGo to User key share');
+    }
+
+    const bitgoPublicGpgKey = await this.getBitgoPublicGpgKey();
+
+    const backupToUserNShare = await encryptNShareFromAPIKeyShare(backupToUserShare, userGpgKey.publicKey);
```

Comment:
This should already be encrypted, right? I don't understand why we need to encrypt it here. I expect that the backup provider would take care of the encryption.